include/nil/blueprint/parser.hpp

```diff
@@ -87,9 +87,38 @@
 
 #include <nil/blueprint/policy/policy_manager.hpp>
 
+#include <nil/blueprint/components/systems/snark/plonk/placeholder/fri_lin_inter.hpp>
+#include <nil/blueprint/recursive_prover/fri_cosets.hpp>
+#include <nil/blueprint/recursive_prover/gate_arg_verifier.hpp>
+#include <nil/blueprint/recursive_prover/permutation_arg_verifier.hpp>
+#include <nil/blueprint/recursive_prover/lookup_arg_verifier.hpp>
+#include <nil/blueprint/recursive_prover/fri_array_swap.hpp>
+
 namespace nil {
     namespace blueprint {
 
+        bool check_operands_constantness(const llvm::CallInst *inst, std::vector<std::size_t> constants_positions) {
+            bool is_const;
+            for (std::size_t i = 0; i < inst->getNumOperands() - 1; i++) {
+                is_const = false;
+                for (std::size_t j = 0; j < constants_positions.size(); j++) {
+                    if(i == constants_positions[j]) {
+                        ASSERT(constants_positions[j] < inst->getNumOperands() - 1);
+                        is_const = true;
+                        if(!llvm::isa<llvm::Constant>(inst->getOperand(i))) {
+                            std::cerr << "\noperand " << i << " must be constant, but it is not\n";
+                            return false;
+                        }
+                    }
+                }
+                if(!is_const && llvm::isa<llvm::Constant>(inst->getOperand(i))) {
+                    std::cerr << "\noperand " << i << " must NOT be constant, but it is\n";
```

Comment:
Why couldn't an arbitrary argument be a constant?

Reply:
We don't put constants into assignment table, but components need them in the table

Reply:
Well, let's put them into the table in this case

Reply:
The whole idea was to use constants as a way to pass variables to intrinsic without writing them into table. Some args must be in the table ( =must not be constant). Some args must not be in the table ( =must be constant).  

For example, this is okay:
`var input;
const int parameter;
__some_intrinsic(input, parameter);`

And this two variants are not okay and will cause problems:
`const var input;
const int parameter;
__some_intrinsic(input, parameter);`
`const var input;
int parameter;
__some_intrinsic(input, parameter);`

I check that every argument that must be constant is constant and vice versa.